lib/forwarder.js

```diff
@@ -36,20 +43,40 @@ export async function forwardToDaemon(resolver, config) {
       let chunk = '';
       while ((chunk = socket.read()) !== null) {
         content += chunk;
-        if (content.length > 5) {
+        if (!fixToStdout && content.length > 5) {
           process.stdout.write(content.substring(0, content.length - 5));
           content = content.substring(content.length - 5);
         }
       }
     })
     .on('end', () => {
-      if (content.startsWith('EXIT')) {
-        process.exitCode = Number(content.slice(4));
-      } else {
+      const endOfContent = content.slice(-5);
+
+      if(!endOfContent.startsWith("EXIT")) {
         process.stdout.write(content);
         console.error('eslint_d: unexpected response');
         process.exitCode = 1;
+        return;
       }
+
+      content = content.slice(0, -5);
+      const exitCode = Number(endOfContent.slice(4));
+      process.exitCode = exitCode;
+
+      if(exitCode !== 0 && exitCode !== 1) {
+        process.stdout.write(content);
+        return;
+      }
+
+      if(!fixToStdout) {
+        return;
+      }
+
+      const object = JSON.parse(content);
+
+      const output = object[0].output || text;
```

Comment:
Does this mean that it's a requirement to use `--fix-to-stdout` together with `--stdin`? If yes, there should be a check for that.

Reply:
This is not a requirement of `eslint` which works fine with and without `--stdin`. This is a requirement of the API I want. When there is no change to do, eslint returns no output. Nevertheless, I think it makes sense to always return an output even if identical to the input.

What do you think?

Reply:
Yes, let's always return the input at least. If the new option is used without `--stdin` it should probably fail with a meaningful error message.

Reply:
I added a check and a corresponding unit test.